modules/common/src/main/java/org/opencastproject/mediapackage/MediaPackageElementFlavor.java

```diff
@@ -392,52 +254,27 @@ public MediaPackageElementFlavor unmarshal(String str) throws Exception {
   }
 
   public boolean matches(MediaPackageElementFlavor other) {
-    if (other == null)
-      return false;
-    if (this == other)
-      return true;
-    if (subtype == null) {
-      if (other.subtype != null && !isWildcard(other.subtype))
-        return false;
-    } else if (!subtype.equals(other.subtype) && (!isWildcard(subtype) && !isWildcard(other.subtype)))
-      return false;
-    if (type == null) {
-      if (other.type != null && !isWildcard(other.type))
-        return false;
-    } else if (!type.equals(other.type) && (!isWildcard(type) && !isWildcard(other.type)))
-      return false;
-    return true;
+    return (other != null) && matchParts(type, other.type) && matchParts(subtype, other.subtype);
+  }
+
+  private boolean matchParts(String part1, String part2) {
+    return part1.equals(part2) || isWildcard(part1) || isWildcard(part2);
   }
 
   @Override
   public int hashCode() {
     final int prime = 31;
     int result = 1;
-    result = prime * result + ((subtype == null) ? 0 : subtype.hashCode());
-    result = prime * result + ((type == null) ? 0 : type.hashCode());
+    result = prime * result + subtype.hashCode();
+    result = prime * result + type.hashCode();
     return result;
   }
 
   @Override
-  public boolean equals(Object obj) {
-    if (this == obj)
-      return true;
-    if (obj == null)
-      return false;
-    if (!(obj instanceof MediaPackageElementFlavor))
-      return false;
-    MediaPackageElementFlavor other = (MediaPackageElementFlavor) obj;
-    if (subtype == null) {
-      if (other.subtype != null)
-        return false;
-    } else if (!subtype.equals(other.subtype))
-      return false;
-    if (type == null) {
-      if (other.type != null)
-        return false;
-    } else if (!type.equals(other.type))
-      return false;
-    return true;
+  public boolean equals(Object other) {
+    return (other != null) && (other instanceof MediaPackageElementFlavor)
```

Comment:
`instanceof` should already take care of the null check:

```suggestion
    return other instanceof MediaPackageElementFlavor
```

Reply:
Thanks, didn't know that.